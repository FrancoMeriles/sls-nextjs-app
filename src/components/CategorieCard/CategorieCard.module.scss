.CategorieCard {
  background-color: white;
  padding: 1.2rem;
  text-decoration: none;
  border-radius: 5px;
  transition: box-shadow 0.3s ease;
  border-color: transparent;
  cursor: pointer;
  min-height: 130px;
  width: 100%;
  position: relative;
  h3 {
    margin: 0 0 1rem 0;
    font-size: 1.1rem;
  }
  span {
    width: 10px;
    height: 10px;
    border-radius: 15px;
    position: absolute;
    bottom: 10px;
    right: 10px;
  }

  &:hover,
  &:focus,
  &:active {
    box-shadow: 0 7px 16px 0 rgb(0 0 0 / 10%), 0 1px 3px 0 rgb(0 0 0 / 10%);
  }
}
