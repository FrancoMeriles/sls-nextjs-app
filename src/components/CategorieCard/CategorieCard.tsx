import React from 'react'
import { LinkBox, LinkOverlay } from '@chakra-ui/react'
import classes from './CategorieCard.module.scss'
import { getRandomColor } from '@base/utils'

const CategorieCard = ({ id, name }) => {
  return (
    <LinkBox className={classes.CategorieCard}>
      <LinkOverlay href={`categorie/${id}`}>
        <h3>{name}</h3>
        <span
          style={{
            backgroundColor: getRandomColor(),
          }}
        />
      </LinkOverlay>
    </LinkBox>
  )
}

export default CategorieCard
