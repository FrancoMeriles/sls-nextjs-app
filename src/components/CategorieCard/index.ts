import CategorieCard from './CategorieCard'

export default CategorieCard
