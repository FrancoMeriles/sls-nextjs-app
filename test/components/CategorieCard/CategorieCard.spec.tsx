import { render, screen } from '@testing-library/react'
import CategorieCard from '@base/components/CategorieCard'
import renderer from 'react-test-renderer'

jest.mock('@base/utils', () => ({
  getRandomColor: () => 'hsl(214,100%,50%)',
}))

const mockCategorieCardProps = {
  id: '__ID__',
  name: '__NAME__',
}

describe('CategorieCard', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer.create(<CategorieCard {...mockCategorieCardProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    render(<CategorieCard {...mockCategorieCardProps} />)
    expect(screen.getByText(mockCategorieCardProps.name)).toBeInTheDocument()
    expect(screen.getByText(mockCategorieCardProps.name).closest('a')).toHaveAttribute(
      'href',
      `/categorie/${mockCategorieCardProps.id}/products?page=1`
    )
  })
})
