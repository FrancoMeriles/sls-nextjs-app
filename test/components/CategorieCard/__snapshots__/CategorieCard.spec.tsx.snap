// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorieCard should render and match snapshot with required props 1`] = `
<div
  className="chakra-linkbox css-15wvuzs"
>
  <a
    className="chakra-linkbox__overlay css-1hnz6hu"
    href="/categorie/__ID__/products?page=1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <p
      className="chakra-text css-pelz90"
    >
      __NAME__
    </p>
    <span
      className="css-1ksdfwg"
    />
  </a>
</div>
`;
